docs/DeveloperGuide.md

```diff
@@ -1,12 +1,220 @@
+if requested again).
+
+How the API component of data fetching works
+1. The initialisation of the class will make an instance of the `Storage`, `HttpClient` and `Ui` class.
+2. If the program just started, the Parking class will call the `loadApiKey` method.
+   - If no file called `secret.txt` is found, file will be created.
+   - If the file is empty, default key will be loaded to the Api instance (but not written to the local file).
+3. To get a certain data set from the Api Service,
+   - Call `asyncExecuteRequest` method to construct the HTTP request packet header and sends the request
+     asynchronously.
+   - Call `fetchData` method to get response immediately from the Api.
+      - This method will also validate the response from the Api by identifying the response code.
+      - Unless its 401 Unauthorised Access (due to invalid Api key), the method will try to fetch data
+        at most 5 times.
+
+Since each API call only returns 500 data, we need to make multiple api calls. Thus, there is another function 
+that handles the API calls all at once. The process is still similar.
+1. Same step 1 and 2 above.
+2. Call `syncFetchData` method, which calls both the `asyncExecuteRequest` and `fetchData` five times (LTA only has 2312 
+Parking lot data)
+   - This method makes all 5 request asynchronously and receives in sequence.
+   - The method will concatenate the data together for data processing by other parts of the program.
+   - The processed data will then be stored in a local data file.
+   
+The following sequence diagram shows how data is fetched using the `update` command.
+![Update Data API](images/UpdateApiSequenceDiagram.png)
+
+
+The API component is also able to:
+- Loads in key from a local file storage (in txt format).
+- Authenticate user API key. If no user key inputted, default key will be loaded.
+- Get API authentication status.
+
+The following sequence diagram shows how the API key is loaded.
+![Api Loading Sequnce Diagram](images/LoadApiSequenceDiagram.png)
+
+
+#### Storage Component
+
+#### Implementation
+![Sequence Diagram](images/LoadFileSequenceDiagram.png)
+##### FileWriter
+
+##### FileLoader
+
+#### UI Component
+
+The user-facing parts of the program are implemented with the Ui class.
+
+It implements the following main functions:
+
+- print() - Prints a string to System.out.
+- getLine() - Asks user for input and returns input.
+- getCommand() - Asks user for a command input and returns input.
+- showLogo() - Prints ASCII art of a car and parKING logo.
+- printResult() - Shows the result of a command execution to the user. Includes additional formatting of the results of 
+different commands.
+- printError() - Print exception message.
+- changeScanner() - Changes the scanner for the Ui object. To be used for JUnit testing.
+- getSeparatorString() - Returns a separator string.
+
+#### Logic Component
```

Comment:
![image](https://user-images.githubusercontent.com/12421064/198495916-81463bd5-d57f-41ef-9a3e-9338d345726c.png)

unclear what is XYZCommand in the diagram, not described in your description as well.